Sources/Orbit/Components/Alert.swift

```diff
@@ -201,55 +193,54 @@ public struct Alert<Content: View>: View {
     }
 
     var isInlineHeaderEmpty: Bool {
-        isIconEmpty && title.isEmpty
-    }
-
-    private var isIconEmpty: Bool {
-        icon?.isEmpty ?? true
+        icon.isEmpty && title.isEmpty
     }
 }
 
 // MARK: - Inits
 public extension Alert {
-    
-    /// Creates Orbit Alert component.
+
+    /// Creates Orbit Alert component with custom content and icon.
     ///
     /// - Parameters:
     ///   - style: A visual style of component. A `status` style can be optionally modified using `status()` modifier when `nil` value is provided.
     init(
         _ title: String = "",
         description: String = "",
-        icon: Icon.Content? = nil,
-        buttons: AlertButtons = .none,
+        buttons: AlertButtons? = nil,
         style: AlertStyle = .status(nil, isSubtle: false),
-        @ViewBuilder content: () -> Content = { EmptyView() }
+        @ViewBuilder content: () -> Content,
+        @ViewBuilder icon: () -> Icon = { EmptyView() }
     ) {
         self.title = title
         self.description = description
-        self.icon = icon
+        self.icon = icon()
         self.buttons = buttons
         self.style = style
-        self.variant = .default
         self.content = content()
     }
 
-    /// Creates Orbit Alert (inline) component.
+    /// Creates Orbit Alert component.
     ///
     /// - Parameters:
     ///   - style: A visual style of component. A `status` style can be optionally modified using `status()` modifier when `nil` value is provided.
     init(
         _ title: String = "",
-        icon: Icon.Content? = nil,
-        button: Button.Content? = nil,
+        description: String = "",
+        icon: Icon.Symbol? = nil,
+        buttons: AlertButtons? = nil,
         style: AlertStyle = .status(nil, isSubtle: false)
-    ) where Content == EmptyView {
-        self.title = title
-        self.description = ""
-        self.icon = icon
-        self.buttons = button.map { .primary($0) } ?? .none
-        self.style = style
-        self.variant = .inline
-        self.content = EmptyView()
+    ) where Icon == Orbit.Icon, Content == EmptyView {
```

Comment:
Doing that would give us a lot of `ambiguous init` errors. Inspired by SwiftUI, we should follow the same principles:

- no matter how many customizable ViewBuilder parameters the component has, they should all use ViewBuilder in a single initializer, not a mix of the other default types (to avoid param combination explosion)
- The initializer that takes the custom ViewBuilder content paramters can only have optional those properties that are already treated by the "simpler" initializer (especially when that parameter is optional even in the simpler initilizer as here - it is less issue in non-optional, such as in case of BarButton)

<img width="770" alt="image" src="https://github.com/kiwicom/orbit-swiftui/assets/35844477/4586b3a0-3ac4-4976-8d8c-68613c9e560f">